diff --git a/imconfig.h b/imconfig.h
index d556cba..1db735f 100644
--- a/imconfig.h
+++ b/imconfig.h
@@ -14,6 +14,8 @@
 
 #pragma once
 
+#define IMGUI_IMPL_OPENGL_ES3
+
 //---- Define assertion handler. Defaults to calling assert().
 // If your macro uses multiple statements, make sure is enclosed in a 'do { .. } while (0)' block so it can be used as a single statement.
 //#define IM_ASSERT(_EXPR)  MyAssert(_EXPR)
