diff --git a/CMakeLists.txt b/CMakeLists.txt
index 26fe83205..ce9072758 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1295,7 +1295,7 @@ elseif(UNIX AND NOT APPLE AND NOT ANDROID AND NOT RISCOS AND NOT HAIKU)
         set(HAVE_FCITX TRUE)
       endif()
 
-      pkg_search_module(IBUS ibus-1.0 ibus)
+      #pkg_search_module(IBUS ibus-1.0 ibus)
       if(IBUS_FOUND)
         set(HAVE_IBUS_IBUS_H TRUE)
         include_directories(${IBUS_INCLUDE_DIRS})
