diff -pruN queue-it/main.py queue-it.patch/main.py
--- queue-it/main.py	2026-09-26 17:15:38
+++ queue-it.patch/main.py	2026-09-26 17:22:06
@@ -107,7 +107,7 @@ DOORS_DICT = {
     (270,64):[0,(16,12),"Main Menu"],
 }
 
-DIALOG_1 = Dialog([("Sign","Welcome to Queue It!"),("Sign","You can move with Q/D or use the left\ngamepad joystick."),("Sign","If you are blocked restart the level\nwith R or the top gamepad button."),("Sign","You can interact with objects by\npressing E or the left gamepad button."),("Sign", "In this game gems go into a queue.\nPress Space or the bottom gamepad\nbutton to use the oldest one first."),("Sign","Each gem gives you a different power,\ngo ahead try the green one.")], 32, 12, 12, True, 12, True, 10, 10)
+DIALOG_1 = Dialog([("Sign","Welcome to Queue It!"),("Sign","You can move with Q/D or use the left\ngamepad joystick or the dpad."),("Sign","If you are blocked restart the level\nwith R or the top gamepad button."),("Sign","You can interact with objects by\npressing E or the left gamepad button."),("Sign", "In this game gems go into a queue.\nPress Space or the bottom gamepad\nbutton to use the oldest one first."),("Sign","Each gem gives you a different power,\ngo ahead try the green one.")], 32, 12, 12, True, 12, True, 10, 10)
 DIALOG_2 = Dialog([("Sign","The blue gem allows you to dash."),("Sign","During that dash, you are invincible.\nGreat for going through spikes."),("Sign","But you can't change direction while\ndashing, so be careful.")], 32, 9, 9, True, 9, True, 10, 10)
 DIALOG_3 = Dialog([("Sign","The pink gem gives you the power to\nphase upward."),("Sign","The phasing is always exactly 4 tiles.\nIf you can't phase there, you can't\nuse the gem.")], 32, 28, 28, True, 28, True, 10, 10)
 DIALOG_4 = Dialog([("Sign","The red gem reverse your gravity."),("Sign","While reversed you jump and phase in\nthe opposite vertical direction.")], 32, 23, 23, True, 23, True, 10, 10)
@@ -521,25 +521,25 @@ def left()-> bool:
     return l
 
 def left()-> bool:
-    return pyxel.btn(pyxel.KEY_Q) or pyxel.btn(pyxel.KEY_A) or pyxel.btn(pyxel.KEY_LEFT) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTX) < -8000
+    return pyxel.btn(pyxel.KEY_Q) or pyxel.btn(pyxel.KEY_A) or pyxel.btn(pyxel.KEY_LEFT) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTX) < -8000 or pyxel.btn(pyxel.GAMEPAD1_BUTTON_DPAD_LEFT)
 
 def right()-> bool:
-    return pyxel.btn(pyxel.KEY_D) or pyxel.btn(pyxel.KEY_RIGHT) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTX) > 8000
+    return pyxel.btn(pyxel.KEY_D) or pyxel.btn(pyxel.KEY_RIGHT) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTX) > 8000 or pyxel.btn(pyxel.GAMEPAD1_BUTTON_DPAD_RIGHT)
 
 def up()-> bool:
-    return pyxel.btn(pyxel.KEY_Z) or pyxel.btn(pyxel.KEY_W) or pyxel.btn(pyxel.KEY_UP) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTY) < -20000
+    return pyxel.btn(pyxel.KEY_Z) or pyxel.btn(pyxel.KEY_W) or pyxel.btn(pyxel.KEY_UP) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTY) < -20000 or pyxel.btn(pyxel.GAMEPAD1_BUTTON_DPAD_UP)
 
 def down()-> bool:
-    return pyxel.btn(pyxel.KEY_S) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTY) > 8000
+    return pyxel.btn(pyxel.KEY_S) or pyxel.btnv(pyxel.GAMEPAD1_AXIS_LEFTY) > 8000 or pyxel.btn(pyxel.GAMEPAD1_BUTTON_DPAD_DOWN)
 
 def use()-> bool:
-    return pyxel.btnp(pyxel.KEY_SPACE) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_A)
+    return pyxel.btnp(pyxel.KEY_SPACE) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_B)
 
 def interact()-> bool:
-    return pyxel.btnp(pyxel.KEY_E) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_X)
+    return pyxel.btnp(pyxel.KEY_E) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_Y)
 
 def restart()-> bool:
-    return pyxel.btnp(pyxel.KEY_R) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_Y)
+    return pyxel.btnp(pyxel.KEY_R) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_X)
 
 def get_room_uv(id:int)-> tuple:
     u = (id % 8) * 36 + 1
@@ -587,7 +587,7 @@ class Game:
         ]
 
         #? Pyxel Manager
-        self.pyxel_manager = PyxelManager((280, 176), scenes, 0, 60, False, False, "assets.pyxres", PALETTE, camera_x=8, camera_y=8)
+        self.pyxel_manager = PyxelManager((280, 176), scenes, 0, 60, True, False, "assets.pyxres", PALETTE, camera_x=8, camera_y=8)
 
         #? Game Variables
         self.id = 0
diff -pruN queue-it/utils.py queue-it.patch/utils.py
--- queue-it/utils.py	2026-08-12 13:18:00
+++ queue-it.patch/utils.py	2026-09-26 17:18:24
@@ -626,7 +626,7 @@ class DialogManager:
 
             if self.__open:
                 if self.__char_index < len(self.__dialog.lines[self.__current_line][1]):
-                    if pyxel.btnp(pyxel.KEY_E) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_X):
+                    if pyxel.btnp(pyxel.KEY_E) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_Y):
                         self.__char_index = len(self.__dialog.lines[self.__current_line][1])
                         if self.__dialog.sound:
                             pyxel.play(self.__dialog.channel, self.__dialog.sound_number)
@@ -636,7 +636,7 @@ class DialogManager:
                             pyxel.play(self.__dialog.channel, self.__dialog.sound_number)
                         self.__char_index += 1
                 else:
-                    if pyxel.btnp(pyxel.KEY_E) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_X):
+                    if pyxel.btnp(pyxel.KEY_E) or pyxel.btnp(pyxel.GAMEPAD1_BUTTON_Y):
                         if self.__current_line < len(self.__dialog.lines) - 1:
                             self.__current_line += 1
                             self.__char_index = 0
