diff --git a/src/raylib.lisp b/src/raylib.lisp
index bd0fdac..7a11ff4 100644
--- a/src/raylib.lisp
+++ b/src/raylib.lisp
@@ -1061,6 +1061,7 @@
 ;;// AudioStream, custom audio stream
 ;;typedef struct AudioStream {
 ;;    rAudioBuffer *buffer;       // Pointer to internal data used by the audio system
+;;    rAudioProcessor *processor; // Pointer to internal data processor, useful for audio effects
 ;;
 ;;    unsigned int sampleRate;    // Frequency (samples per second)
 ;;    unsigned int sampleSize;    // Bit depth (bits per sample): 8, 16, 32 (24 not supported)
@@ -1069,20 +1070,22 @@
 (defcstruct (%audio-stream :class audio-stream-type)
  "Audio stream type"
   (buffer :pointer)
+  (processor :pointer)
   (sample-rate :unsigned-int)
   (sample-size :unsigned-int)
   (channels :unsigned-int))
 
 (defmethod translate-into-foreign-memory (object (type audio-stream-type) pointer)
- (with-foreign-slots ((buffer sample-rate sample-size channels) pointer (:struct %audio-stream))
+ (with-foreign-slots ((buffer processor sample-rate sample-size channels) pointer (:struct %audio-stream))
                       (setf buffer (nth 0 object))
-                      (setf sample-rate (nth 1 object))
-                      (setf sample-size (nth 2 object))
-                      (setf channels (nth 3 object))))
+                      (setf processor (nth 1 object))
+                      (setf sample-rate (nth 2 object))
+                      (setf sample-size (nth 3 object))
+                      (setf channels (nth 4 object))))
 
 (defmethod translate-from-foreign (pointer (type audio-stream-type))
-  (with-foreign-slots ((buffer sample-rate sample-size channels) pointer (:struct %audio-stream))
-                      (list buffer sample-rate sample-size channels)))
+  (with-foreign-slots ((buffer processor sample-rate sample-size channels) pointer (:struct %audio-stream))
+                      (list buffer processor sample-rate sample-size channels)))
 
 ;;// Sound
 ;;typedef struct Sound {
