diff --git a/configure.ac b/configure.ac
index 960b68cd..d480b784 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,7 +42,6 @@ AC_PROG_CC
 AC_PROG_RANLIB
 AC_PROG_LN_S
 PKG_PROG_PKG_CONFIG
-AX_CXX_COMPILE_STDCXX_11(,[mandatory])
 save_CXXFLAGS="$CXXFLAGS"
 
 GLOBAL_CFLAGS="-include \"\$(top_builddir)/include/config.h\""
diff --git a/src/OVGA.cpp b/src/OVGA.cpp
index f5fdca5c..866823d3 100644
--- a/src/OVGA.cpp
+++ b/src/OVGA.cpp
@@ -119,7 +119,7 @@ int Vga::init()
 
    SDL_WarpMouseGlobal(mouse_x, mouse_y); // warp to initialize mouse by event queue
 
-   Uint32 window_pixel_format = SDL_GetWindowPixelFormat(window);
+   Uint32 window_pixel_format = SDL_PIXELFORMAT_ARGB8888; //SDL_GetWindowPixelFormat(window);
    if (window_pixel_format == SDL_PIXELFORMAT_UNKNOWN)
    {
       ERR("Unknown pixel format: %s\n", SDL_GetError());
