diff --git a/engine/qclib/initlib.c b/engine/qclib/initlib.c
index 70c36a9af..a837cd67b 100644
--- a/engine/qclib/initlib.c
+++ b/engine/qclib/initlib.c
@@ -552,8 +552,9 @@ static void PDECL PR_Configure (pubprogfuncs_t *ppf, size_t addressable_size, in
 		addressable_size = 32*1024*1024;
 #endif
 	}
-	if (addressable_size > 0x80000000)
-		addressable_size = 0x80000000;
+        #define MAX_ADDRESSABLE_SIZE (100 * 1024 * 1024)  // 100MB
+        if (addressable_size > MAX_ADDRESSABLE_SIZE)
+                addressable_size = MAX_ADDRESSABLE_SIZE;
 	PRAddressableFlush(progfuncs, addressable_size);	
 	progfuncs->funcs.stringtable = prinst.addressablehunk;
 
