diff --git a/src/graphics/font/Font.cpp b/src/graphics/font/Font.cpp
index 26e7d3cd2..cc75d9498 100644
--- a/src/graphics/font/Font.cpp
+++ b/src/graphics/font/Font.cpp
@@ -365,7 +365,7 @@ Font::TextSize Font::process(int x, int y, std::string_view text, Color color) {
 		m_textures->upload();
 		
 		UseRenderState state(render2D());
-		UseTextureState textureState(TextureStage::FilterNearest, TextureStage::WrapClamp);
+		UseTextureState textureState(TextureStage::FilterLinear, TextureStage::WrapClamp);
 		
 		// Fixed pipeline texture stage operation
 		GRenderer->GetTextureStage(0)->setColorOp(TextureStage::OpDisable);
