diff --git a/sources/main.cpp b/sources/main.cpp
index 352d961..d0e09b8 100644
--- a/sources/main.cpp
+++ b/sources/main.cpp
@@ -49,24 +49,28 @@
 
 char *application_name = "F-1 Spirit";
 int application_version = 0;
-#ifdef PANDORA
-int SCREEN_X = 800;
-#elif defined(PYRA)
-int SCREEN_X = 1280;
-#else
-int SCREEN_X = 640;
-#endif
-#ifdef PYRA
-int SCREEN_Y = 720;
-#else
-int SCREEN_Y = 480;
-#endif
+// #ifdef PANDORA
+// int SCREEN_X = 800;
+// #elif defined(PYRA)
+// int SCREEN_X = 1280;
+// #else
+// int SCREEN_X = 640;
+// #endif
+// #ifdef PYRA
+// int SCREEN_Y = 720;
+// #else
+// int SCREEN_Y = 480;
+// #endif
+long SCREEN_X = 640;
+long SCREEN_Y = 480;
+bool fullscreen = false;
+
 int g_stencil_bits = 0;
 int N_SFX_CHANNELS = 16;
 int COLOUR_DEPTH = 32;
 int MAX_CONNECTIONS = 16;
 bool sound = true;
-bool fullscreen = false;
+// bool fullscreen = false;
 bool network = true;
 int network_tcp_port = 32124;
 int network_udp_port = 32125;
@@ -158,11 +162,11 @@ SDL_Surface *initialization(int flags)
 
 #endif
 #ifdef HAVE_GLES
-	fullscreen = true;
+	// fullscreen = true;
 	flags = SDL_FULLSCREEN;
 #else
 #ifdef PYRA
-        fullscreen = true;
+        // fullscreen = true;
         flags = SDL_FULLSCREEN;
 #endif
 	flags = SDL_OPENGL | flags;
@@ -294,13 +298,33 @@ int main(int argc, char** argv) {
 	bool quit = false;
 	bool need_to_redraw = true;
 
+	char* endptr;
+
+    char* envwidth = getenv("DISPLAY_WIDTH");
+	if (envwidth != NULL)
+		SCREEN_X = strtol(envwidth, &endptr, 10);
+
+    char* envheight = getenv("DISPLAY_HEIGHT");
+	if (envheight != NULL)
+		SCREEN_Y = strtol(envheight, &endptr, 10);
+
+    char* envfullscreen = getenv("FULLSCREEN");
+    if (envfullscreen != NULL) {
+        if (strcmp(envfullscreen, "true") == 0 || strcmp(envfullscreen, "TRUE") == 0)
+            fullscreen = true;
+    }
+
+    printf("Display width = %ld\n", SCREEN_X); 
+    printf("Display height = %ld\n", SCREEN_Y);
+    printf("Fullscreen = %d\n", fullscreen); 
+
 #ifdef F1SPIRIT_DEBUG_MESSAGES
 
 	output_debug_message("Application started\n");
 #endif
-#ifdef HAVE_GLES
-	fullscreen = true;
-#endif
+// #ifdef HAVE_GLES
+// 	fullscreen = true;
+// #endif
 	screen_sfc = initialization((fullscreen ? SDL_FULLSCREEN : 0));
 
 	if (screen_sfc == 0)
