diff --git a/src/main.cpp b/src/main.cpp
index 1ef4306..3e6119c 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -217,7 +221,7 @@ void createDefaultConfigFile(const std::string& configfilepath, const std::strin
                                 "Play Intro = false          # Play the intro when starting the game?\n"
                                 "Player Name = %s            # The name of the player\n"
                                 "Language = %s               # en = English, fr = French, de = German\n"
-                                "Scroll Speed = 50           # Amount to scroll the map when the cursor is near the screen border\n"
+                                "Scroll Speed = 15           # Amount to scroll the map when the cursor is near the screen border\n"
                                 "Show Tutorial Hints = true  # Show tutorial hints during the game\n"
                                 "\n"
                                 "[Video]\n"
@@ -502,7 +506,7 @@ int main(int argc, char *argv[]) {
             settings.general.playIntro = myINIFile.getBoolValue("General","Play Intro",false);
             settings.general.playerName = myINIFile.getStringValue("General","Player Name","Player");
             settings.general.language = myINIFile.getStringValue("General","Language","en");
-            settings.general.scrollSpeed = myINIFile.getIntValue("General","Scroll Speed",50);
+            settings.general.scrollSpeed = myINIFile.getIntValue("General","Scroll Speed",15);
             settings.general.showTutorialHints = myINIFile.getBoolValue("General","Show Tutorial Hints",true);
             settings.video.width = myINIFile.getIntValue("Video","Width",640);
             settings.video.height = myINIFile.getIntValue("Video","Height",480);
