diff --git a/src/misc/FileSystem.cpp b/src/misc/FileSystem.cpp
index 4d4b9bf..6aef274 100644
--- a/src/misc/FileSystem.cpp
+++ b/src/misc/FileSystem.cpp
@@ -408,9 +408,9 @@ std::string getDuneLegacyDataDir() {
     if(duneLegacyDataDir.empty()) {
 
         std::string dataDir;
-#ifdef DUNELEGACY_DATADIR
-        dataDir = DUNELEGACY_DATADIR;
-#endif
+// #ifdef DUNELEGACY_DATADIR
+//         dataDir = DUNELEGACY_DATADIR;
+// #endif
 
         if((dataDir.empty()) || (dataDir == ".") || (dataDir == "./") || (dataDir == ".\\")) {
             char* basePath = SDL_GetBasePath();
