--- src/main.cpp.orig	2025-03-05 06:00:17.965188632 +0000
+++ src/main.cpp	2025-03-05 06:00:20.870474962 +0000
@@ -156,7 +156,7 @@
     SDL_RenderSetLogicalSize(renderer, settings.video.width, settings.video.height);
     screenTexture = SDL_CreateTexture(renderer, SCREEN_FORMAT, SDL_TEXTUREACCESS_TARGET, settings.video.width, settings.video.height);
 
-    SDL_ShowCursor(SDL_DISABLE);
+    // SDL_ShowCursor(SDL_DISABLE);
 }
 
 void toogleFullscreen()
