--- /home/me/Desktop/PlanetD4RK_WindowsV2/PlanetD4RK_WindowsV2/conf.lua	2022-02-23 12:26:36.000000000 +1100
+++ /home/me/Desktop/gt/PortMaster-New/ports/planetd4rk/planetd4rk/PlanetD4RK_WindowsV2.love/conf.lua	2024-11-01 19:20:54.000000000 +1100
@@ -21,7 +21,8 @@
     t.window.title = "PLANET D4RK"
     t.window.width = windowWidth
     t.window.height = windowHeight
+t.window.resizable = true  -- !!! MAKE WINDOW RESIZABLE
     if web and not itch then
         t.modules.joystick = false
     end
-end
\ No newline at end of file
+end
