diff --git a/source/Main.hx b/source/Main.hx
index cb1c077..c99c224 100644
--- a/source/Main.hx
+++ b/source/Main.hx
@@ -15,7 +15,7 @@ class Main extends Sprite
 	#if web
 	var framerate:Int = 60;
 	#else
-	var framerate:Int = 144;
+	var framerate:Int = 60;
 	#end
 
 	public function new()
@@ -27,7 +27,7 @@ class Main extends Sprite
 		Application.current.window.element.style.setProperty("image-rendering", "pixelated");
 		#end
 
-		addChild(new FlxGame(480, 160, () -> new Splash(MenuState.new), framerate, framerate, true));
+		addChild(new FlxGame(380, 160, () -> new Splash(MenuState.new), framerate, framerate, true));
 
 		Lib.current.stage.addEventListener(openfl.events.KeyboardEvent.KEY_DOWN, (e:KeyboardEvent) ->
 		{
diff --git a/source/MenuState.hx b/source/MenuState.hx
index 539c564..0876440 100644
--- a/source/MenuState.hx
+++ b/source/MenuState.hx
@@ -60,7 +60,7 @@ class MenuState extends FlxState
 		// somewhat gracefully fallback if there's no API keys stuff
 
 		NG.createAndCheckSession(API.ngapi, true, null);
-		NG.core.setupEncryption(API.ngenc);
+//		NG.core.setupEncryption(API.ngenc);
 
 		var notifString:String = "Connected to Newgrounds";
 
@@ -72,10 +72,10 @@ class MenuState extends FlxState
 				notifString += ". Press N to login";
 		}
 
-		new FlxTimer().start(4, _ ->
-		{
-			Notification.instance.genTexts(notifString, 4, BOTTOM_LEFT);
-		});
+//		new FlxTimer().start(4, _ ->
+//		{
+//			Notification.instance.genTexts(notifString, 4, BOTTOM_LEFT);
+//		});
 	}
 
 	function attemptLogin()
