--- boswars2/fabricate.py	2024-09-01 17:46:19.776593866 +0000
+++ boswars/fabricate.py	2024-08-30 18:01:53.943668810 +0000
@@ -711,7 +711,7 @@
         return False
 
 # default Builder instance, used by helper run() and main() helper functions
-default_builder = Builder()
+default_builder = Builder(runner='always_runner')
 default_command = 'build'
 
 def setup(builder=None, default=None, **kwargs):
