diff --git a/src/states/StateEditProfile.cpp b/src/states/StateEditProfile.cpp
index fba5b15c..f6c4e32f 100644
--- a/src/states/StateEditProfile.cpp
+++ b/src/states/StateEditProfile.cpp
@@ -156,7 +156,7 @@ void StateEditProfile::checkEvents() {
                           std::string(GAMETEXT_ENTERPLAYERNAME) + ":",
                           UI_MSGBOX_OK | UI_MSGBOX_CANCEL,
                           true,
-                          "");
+                          "Portmaster");
     v_msgboxState->setMsgBxId("NEWPROFILE");
     StateManager::instance()->pushState(v_msgboxState);
   }
