diff --git a/src/common/XMSession_default.h b/src/common/XMSession_default.h
index dbf58d4a..c955c43c 100644
--- a/src/common/XMSession_default.h
+++ b/src/common/XMSession_default.h
@@ -65,7 +65,7 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #define DEFAULT_TESTTHEME false
 #define DEFAULT_GHOST_MYBEST true
 #define DEFAULT_GHOST_THEBEST false
-#define DEFAULT_GHOSTBESTREFROOM true
+#define DEFAULT_GHOSTBESTREFROOM false
 #define DEFAULT_GHOSTBESTOTHERROOMS false
 #define DEFAULT_AUTOSAVEHIGHSCORESREPLAYS true
 #define DEFAULT_DISABLEANIMATIONS false
diff --git a/src/states/StateMainMenu.cpp b/src/states/StateMainMenu.cpp
index 65cb2507..c49ea43c 100644
--- a/src/states/StateMainMenu.cpp
+++ b/src/states/StateMainMenu.cpp
@@ -201,8 +201,8 @@ void StateMainMenu::enter() {
     /* Should we show a notification box? (with important one-time info) */
     /* thus, is it really required to save this value as profile value ?!? */
     if (XMSession::instance()->notifyAtInit()) {
-      StateManager::instance()->pushState(
-        new StateMessageBox(NULL, GAMETEXT_NOTIFYATINIT, UI_MSGBOX_OK));
+      // StateManager::instance()->pushState(
+      //   new StateMessageBox(NULL, GAMETEXT_NOTIFYATINIT, UI_MSGBOX_OK));
       XMSession::instance()->setNotifyAtInit(false);
     }
 
