diff --git a/signus/src/main.cpp b/signus/src/main.cpp
index ae31218..b292790 100644
--- a/signus/src/main.cpp
+++ b/signus/src/main.cpp
@@ -40,6 +41,11 @@ volatile int signus_running    = FALSE;

 void *Signus_BackupScr = NULL;

+void sigterm_handler(int signal, siginfo_t *info, void *_unused)
+{
+    exit(0);
+}
+

 // Destrukce wokenich objektu:

@@ -113,6 +121,14 @@ static void signus_thread(void *)
 extern int InitGlobal();
 
 int main(int argc, char *argv[]) {
+    struct sigaction action;
+    action.sa_handler = NULL;
+    action.sa_sigaction = sigterm_handler;
+    action.sa_flags = SA_SIGINFO;
+    action.sa_restorer = NULL;
+
+    sigaction(SIGTERM, &action, NULL);
+
 	// Honor system locale
 	setlocale(LC_ALL, "");
 
