diff -ruN -x '*.o' -x '*.a' -x Makefile -x config.h -x config.cflags -x config.ldflags -x makefile.opts -x .config.log -x '.tmp*' -x qdgdf_i -x filp_i -x test -x filp fr2002-orig/filp/config.sh fr2002-19.10.29/filp/config.sh
--- fr2002-orig/filp/config.sh	2026-09-22 12:14:47.000000000 +0000
+++ fr2002-19.10.29/filp/config.sh	2026-10-07 05:42:04.043330504 +0000
@@ -26,6 +26,7 @@
     --without-win32)        WITHOUT_WIN32=1 ;;
     --without-unix-glob)    WITHOUT_UNIX_GLOB=1 ;;
     --without-regex)        WITHOUT_REGEX=1 ;;
+    --without-readline)     WITHOUT_READLINE=1 ;;
     --with-included-regex)  WITH_INCLUDED_REGEX=1 ;;
     --with-pcre)            WITH_PCRE=1 ;;
     --help)                 CONFIG_HELP=1 ;;
@@ -53,6 +54,7 @@
     echo "--prefix=PREFIX       Installation prefix ($PREFIX)."
     echo "--without-win32       Disable win32 interface detection."
     echo "--without-unix-glob   Disable glob.h usage (use workaround)."
+    echo "--without-readline    Disable readline usage."
     echo "--with-included-regex Use included regex code (gnu_regex.c)."
     echo "--with-pcre           Enable PCRE library detection."
     echo "--mingw32             Build using the mingw32 compiler."
@@ -258,6 +260,10 @@
 # test for readline()
 echo -n "Testing for readline()... "
 
+if [ "$WITHOUT_READLINE" = "1" ] ; then
+    echo "Disabled"
+else
+
 echo "#include <readline/readline.h>" > .tmp.c
 echo "#include <readline/history.h>" >> .tmp.c
 echo "int main(void) { readline(\"prompt:\"); return 0; }" >> .tmp.c
@@ -272,6 +278,8 @@
     echo "No"
 fi
 
+fi
+
 # test for Grutatxt
 echo -n "Testing if Grutatxt is installed... "
 
