diff --git a/CMakeLists.txt b/CMakeLists.txt
index 941fc25..3619209 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,7 +161,7 @@ endif(NOT OGGVORBIS_FOUND)
 
 ### SDL
 
-OPTION(AQUARIA_INTERNAL_SDL "Always use included SDL library" ${WIN32_TRUE})
+OPTION(AQUARIA_INTERNAL_SDL "Always use included SDL library" FALSE)
 if(NOT AQUARIA_INTERNAL_SDL)
     if(AQUARIA_USE_SDL2)
         set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
@@ -225,7 +225,7 @@ endif (NOT OPENAL_FOUND)
 
 ### TinyXML2
 
-OPTION(AQUARIA_INTERNAL_TINYXML2 "Always use included TinyXML2 library" ${WIN32_TRUE})
+OPTION(AQUARIA_INTERNAL_TINYXML2 "Always use included TinyXML2 library" TRUE)
 if(NOT AQUARIA_INTERNAL_TINYXML2)
     set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
     find_package(TinyXML2)
